tools/istio-iptables/pkg/log/nflog.go

```diff
@@ -0,0 +1,117 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"context"
+	golog "log"
+	"net"
+	"os"
+	"time"
+
+	"github.com/florianl/go-nflog/v2"
+	"golang.org/x/net/ipv4"
+
+	"istio.io/pkg/env"
+	"istio.io/pkg/log"
+)
+
+var TraceLoggingEnabled = env.RegisterBoolVar(
+	"IPTABLES_TRACE_LOGGING",
+	false,
+	"When enable, all iptables actions will be logged. "+
+		"This requires NET_ADMIN privilege and has noisy logs; as a result, this is intended for debugging only").Get()
+
+var iptablesTrace = log.RegisterScope("iptables", "trace logs for iptables", 0)
+
+// ReadNFLOGSocket reads from the nflog socket, sending output to logs.
+// This is intended for debugging only.
+func ReadNFLOGSocket(ctx context.Context) {
```

Comment:
Is this that could be done by a standalone command or helper ( or iptables ), without bloating pilot-agent ?

Reply:
I thought about making it in istioctl or something but its a bit tricky since it needs to run on the actual node. I think best case for standalone would be ephemeral containers but that isn't really ready unfortunately... It is a bit nice to have the access logs + iptables logs side by side in some cases

Reply:
Given the direction is to reduce permissions and isolate network - I'm not so sure. But it's not a huge dependency, compared with the others.